framework/src/variables/MooseVariableFV.C

```diff
@@ -665,7 +669,8 @@ MooseVariableFV<OutputType>::uncorrectedAdGradSln(const FaceInfo & fi) const
     const VectorValue<ADReal> & elem_two_grad = adGradSln(elem_two);
 
     // Uncorrected gradient value
-    unc_face_grad = Moose::FV::linearInterpolation(elem_one_grad, elem_two_grad, fi);
+    unc_face_grad =
+        Moose::FV::linearInterpolation(elem_one_grad, elem_two_grad, fi, elem_one_is_fi_elem);
```

Comment:
so when the centroid of the face is far from the center of the segment between the two element centroids do we have a correction or still use that

Reply:
I think there are two things:
1. What if one element's centroid is much closer to the face than the other's? This is handled within the linear interpolation; we have a geometric weighting factor for this
2. What if the line connecting the two element centroids doesn't pass through the centroid of the face? This happens on skewed meshes. It is known that if there is no correction for the skew, then you lose the second order convergence of the method. We *do not* currently have a skew correction. Let me open an issue for that

Reply:
Ok, I've created https://github.com/idaholab/moose/issues/16239

Reply:
Nice thanks.
I was talking about 2 but yes, 1 is properly addressed.